test/extended/single_node/topology.go

```diff
@@ -0,0 +1,42 @@
+package single_node
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	exutil "github.com/openshift/origin/test/extended/util"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	"strings"
+)
+
+var _ = Describe("[sig-arch] Cluster topology single node tests", func() {
+	oc := exutil.NewCLIWithoutNamespace("singlenode").Verbose()
+
+	It("Verify that topology API is respected", func() {
+		out, err := oc.Run("get").Args("infrastructure", "cluster",
+			"-o", "template", "--template",
+			`{{.status.controlPlaneTopology}}`,
```

Comment:
@omertuc to make it fit the API definition, I would address also the infstructureTopology. Even if we not sure yet what operator will belong to which part, lets at least verify that they both SingleReplica

@ravidbro WDYT?

Reply:
what are we testing here? that the field exists and populated?
If yes, then InfraTopolgy field should be checked too

Reply:
The test right now checks that if the control plane topology is set to SingleReplica then all relevant deployments take it into account and have replica 1.

@romfreiman is suggesting to prepare the test for a future where we might want to to check some deployments when controlplane is single and other deployments when infra is single. Or maybe just verify that controlplane==infra==single for now

Right now the test automatically skips if the cluster has a HA topology

Reply:
the ingress deployment will have 1 replica depending only on the InfrastructureTopology

Reply:
Noted, thanks